app/models/node.rb

```diff
@@ -102,7 +102,7 @@ def destroy_attachments
   end
 
   def parent_node
-    if self.parent.nil?
+    if self.parent.nil? || self.parent.project.id != self.project.id
```

Comment:
let's avoid unnecessary SQL...

```suggestion
    if self.parent.nil? || self.parent.project_id != self.project_id
```

Reply:
this was like this because `node.project_id` does not exist in CE